manager/state/store/tasks.go

```diff
@@ -224,8 +225,15 @@ func (ti taskIndexerByName) FromObject(obj interface{}) (bool, []byte, error) {
 		panic("unexpected type passed to FromObject")
 	}
 
+	name := t.Annotations.Name
+	if name == "" {
+		// If Task name is not assigned then calculated name is used like before.
+		// This might be removed in the future.
+		name = fmt.Sprintf("%v.%v.%v", t.ServiceAnnotations.Name, t.Slot, t.Task.ID)
```

Comment:
I'm confused - why is this different than the code in `ListTasks`:

```go
name = fmt.Sprintf("%v.%v", e.ServiceAnnotations.Name, e.Slot)
```

If the name in the index doesn't match the name computed in `filterTasks`, this won't work for non-prefix matches.